packages/saved-views-react/src/applySavedView.ts

```diff
@@ -77,6 +94,101 @@ export interface ApplySavedViewSettings {
  */
 type ApplyStrategy = "apply" | "keep";
 
+async function createSeedViewStateProps(
+  iModel: IModelConnection,
+  viewport: Viewport,
+  savedViewData: SavedViewData,
+  settings: ApplySavedViewSettings | undefined = {},
+): Promise<ViewStateProps> {
+  if (settings.viewState !== "keep") {
+    return settings.viewState instanceof ViewState
+      ? settings.viewState.toProps()
+      : createViewStateProps(iModel, savedViewData.viewData);
+  }
+  return viewport.view.toProps();
+}
+
+function applyCameraOptions(
+  seedViewStateProps: ViewStateProps,
+  _iModel: IModelConnection,
+  viewport: Viewport,
+  savedViewData: SavedViewData,
+  settings: ApplySavedViewSettings,
+): ViewStateProps {
+  const setCameraInfo = (
+    seedViewDef: ViewStateProps,
+    cameraProps: ViewPose | ViewDefinitionProps | ViewData,
+    viewDataType: string = "iTwin3d",
+  ): ViewStateProps => {
+    if (viewDataType === "iTwin3d") {
```

Comment:
Just to ensure there are no typos, could this `iTwin3d` (and the other values) be put into an enum or a type? e.g.
`type ViewDataType = "iTwin3d" | "<whatever the one for 2d is>" | ...;`

[Example](https://www.typescriptlang.org/play/?#code/C4TwDgpgBAsiAq5oF4oCIDKB7AthAagIYA2ArhGlAD7oDywAFhAE5FkUDcAsAFC8D0-KAGFcYAJbEWUcQGcoDQmDAgAXLwDGWAHazgUAOZYsAEzblVsBEiipMuAiXJpuPQSLGTpWnGGKFxXXUeLV19ACNCMycISzhESFt0YAg9FyA)

Reply:
That sounds good, as long as the compiler is properly complaining and showing the squigglies if you incorrectly write it up when comparing `viewDataType`, I'm good.